### ISSUE TITLE
Warning: A Component: React.createClass is deprecated

### ISSUE THREAD

**Issue by rinama** [omcljs/om#881]
After upgrading om to version "1.0.0-beta1", the following error is printed in the console (in development mode only)
`react.inc.js:3287 Warning: A Component: React.createClass is deprecated and will be removed in version 16. Use plain JavaScript classes instead. If you're not yet ready to migrate, create-react-class is available on npm as a drop-in replacement.`

**Comment by danielstockton** [comment]
It's coming from [here](https://github.com/omcljs/om/blob/master/src/main/om/dom.cljs#L13).
Other projects e.g. [reagent](https://github.com/reagent-project/reagent/pull/298/files) solved it by depending on `cljsjs/create-react-class` and replacing with a call to `js/createReactClass`.

**Comment by bhauman** [comment]
I would advise working on a solution that extends `React.Component` so that the solution is backwards compatible.

**Comment by wilkerlucio** [comment]
@bhauman the PR that I sent uses the official react fallback for it; I guess this is the safest as we can get.

**Comment by bhauman** [comment]
@wilkerlucio its a valid solution

It has it's limitations though. It depends on another package, which will be probably be deprecated itself much sooner than we want.

It limits users to fairly recent React versions whereas the ability to extend `React.Component` has been around since React 14.

I guess I'm just saying that it's probably time we get onboard the train.

**Comment by wilkerlucio** [comment]
@bhauman not sure what you mean, according to React docs: https://facebook.github.io/react/docs/react-without-es6.html using the `create-react-class` package is the official way for people that don't want to go with ES6, as far as I understand this is not supposed to get deprecated.

**Comment by wilkerlucio** [comment]
@bhauman I mean, using the class syntax would mean having to support it on CLJS somehow, or rely on some ufly hacks with `js*`, do you think this is safer than using the `create-react-class` package?

**Comment by bhauman** [comment]
This is indeed possible. You won't have to change any deps whatsoever if we extend `React.Component`.

**Comment by bhauman** [comment]
I've just worked this out in Devcards I can get you some example code.

**Comment by wilkerlucio** [comment]
I would like to see that, but being honest, I prefer using the official implementation for it, there are plenty of caveats in the `create-react-class` implementation, creating a custom thing has a much higher chance of missing some feature of that IMO.

**Comment by bhauman** [comment]
https://github.com/bhauman/devcards/blob/master/src/devcards/util/utils.clj#L8
https://github.com/bhauman/devcards/blob/master/src/devcards/core.cljs#L287

**Comment by bhauman** [comment]
I hear you, it is a bigger lift in this case to verify that everything is working correctly.

**Comment by bhauman** [comment]
Yeah after looking at it, it is a bit more invasive to make these changes to Om.

**Comment by wilkerlucio** [comment]
Thanks for the links, I'm glad you could make it work with so little. As you said, the change is not so simple in `om` codebase. Unless Facebook says somethign about the `create-react-class` getting deprecated it should be safe for the forseable future.

**Comment by nbardy** [comment]
Any updates on this now that React 16.0 is out. From the looks of it this is all that is required to make om 16.0 compatible.
